.github/renovate.json5

```
@@ -82,6 +82,13 @@
       "matchPackagePatterns": ["kindest\/node"],
       "enabled": false
     },
+    {
+      // Only patch level updates for logcheck. Minor and major versions must be in sync with golangci-lint.
+      "matchDatasources": ["go"],
+      "matchUpdateTypes": ["major", "minor"],
+      "matchFileNames": ["hack\/tools\/logcheck\/go\\.mod"],
+      "enabled": false
+    },
```

```suggestion
    {
      // Go versions must be in sync with golangci-lint.
      "matchDatasources": ["go"],
      "matchFileNames": ["hack\/tools\/logcheck\/go\\.mod"],
      "enabled": false
    },
```
You removed the patch level in `go.mod`, so there can be no patch version update anyway.

---

Hm, aren't there patch updates as of Go 1.22?

---

Yes, there are, but the rule is for updating the `go` statement in `go.mod`. 
This is `go 1.21` in your proposal and does not include any patch version.